app/services/common_platform/api/defendant_finder.rb

```diff
@@ -76,6 +76,15 @@ def detect_spaces_error
                                            :spaces_in_urn)
       end
 
+      def load_from_common_platform(urn)
+        results = CommonPlatform::Api::SearchProsecutionCase.call(prosecution_case_reference: urn)
+
+        # Very occasionally this case-insensitive URN search will return 2 results, and in that
+        # case we prefer the one that matches the URN exactly. Failing that we just take the first
+        # result
+        results.find { _1.body["prosecutionCaseReference"] == urn } || results.first
```

Comment:
Ruby 3.4 introduce `it` as block param, so we can use this now:
`results.find { it.body["prosecutionCaseReference"] }`
which is more readable.

https://www.mintbit.com/blog/writing-more-expressive-ruby-with-the-it-shorthand/

Reply:
it's not clear (at lest to me 😄  ) why
`results.find { it.body["prosecutionCaseReference"] == urn } ` could possibly be `nil ` ?

Reply:
Oh yes, "it" is great! I will change it.

The issue I was worrying about is if `urn` is "TEST12345" but the only result has prosecutionCaseReference "test12345". My worry was that in some cases these are treated as separate URNs but in some they are the same. BUT thinking about it, since the `urn` we use comes from our local copy of Common Platform data, it should _always_ match a Common Platform record, so you're right, it should never be nil.  I've updated the comment and removed the fallback.